src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -177,24 +185,41 @@ public X509Certificate[] getAcceptedIssuers() {
                 }
             }
 
-            // initialize truststore
-            KeyStore ts = KeyStore.getInstance(KeyStore.getDefaultType());
+            KeyStore ts = KeyStore.getInstance(truststoreType);
             ts.load(null, null);
 
-            for (TruststoreConfig truststore : truststores) {
+            // initialize truststore with user provided path and pass
+            if (!truststorePath.isEmpty() && !truststorePass.isEmpty()) {
+                try (InputStream truststore = new FileInputStream(truststorePath.get())) {
+                    ts.load(truststore, truststorePass.get().get(passCharset).toCharArray());
+                    truststorePass.get().clear();
```

Comment:
Move this `.clear()` call into a `finally` clause, please, to ensure it's blanked out even if the truststore fails to load.